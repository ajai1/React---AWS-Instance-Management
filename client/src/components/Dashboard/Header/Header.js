import React from "react";
import { AppBar, Toolbar, Typography } from "@material-ui/core";
import { useHistory } from "react-router-dom";

import axios from "axios";

import "./Header.css";

const Header = () => {
  const history = useHistory();

  const logout = async () => {
    const res = await axios.post("/api/logout", {
      headers: {
        Authorization: "Bearer " + localStorage.getItem("authorization"),
      },
    });
    if (res.data) {
      localStorage.removeItem("authorization");
    }
    console.log(res.data);
    history.push({
      pathname: "/login",
    });
  };

  return (
    <div>
      <AppBar position="static">
        <Toolbar className="header__container">
          <div>
            <Typography variant="h6">Dashboard</Typography>
          </div>

          <div>
            <Typography
              style={{ cursor: "pointer" }}
              variant="body1"
              onClick={logout}
              data-testid="logout"
            >
              Logout
            </Typography>
          </div>
        </Toolbar>
      </AppBar>
    </div>
  );
};

export default Header;
