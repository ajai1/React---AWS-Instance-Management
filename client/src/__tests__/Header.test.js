import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Header from "../components/Dashboard/Header/Header";

afterEach(cleanup);

/*
    To check if the Header is not changed, and a layout is maintained
    To Prevent the changes made to the Header Component
*/
it("renders", () => {
  const { asFragment } = render(<Header />);
  expect(asFragment()).toMatchSnapshot();
});

it("Logout", () => {
  const { getByTestId, getByText } = render(<Header />);
  expect(getByTestId("logout")).toHaveTextContent("Logout");
});
