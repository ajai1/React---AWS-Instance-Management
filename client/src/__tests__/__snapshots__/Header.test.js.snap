// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular header__container MuiToolbar-gutters"
      >
        <div>
          <h6
            class="MuiTypography-root MuiTypography-h6"
          >
            Dashboard
          </h6>
        </div>
        <div>
          <p
            class="MuiTypography-root MuiTypography-body1"
            data-testid="logout"
            style="cursor: pointer;"
          >
            Logout
          </p>
        </div>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
